scanner/rev_cache.go

```diff
@@ -0,0 +1,30 @@
+package scanner
+
+import "sync"
+
+type RevCache struct {
+	m  map[string]string
+	mu *sync.Mutex
+}
+
+func NewRevCache(m map[string]string) *RevCache {
+	return &RevCache{
+		m:  m,
+		mu: &sync.Mutex{},
+	}
+}
+
+func (c *RevCache) GetName(sha string) (string, bool) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	rev, ok := c.m[sha]
+	return rev, ok
+}
+
+func (c *RevCache) Cache(sha, name string) {
```

Comment:
Ha, I totally misunderstood the code when I was reading it for the first time. The `Cache(sha, name string)` caches the SHA1 hash of a file to its name on disk (as used [here](https://github.com/github/git-lfs/blob/e0db9cc16262528c476aa04420461315e8bfca23/lfs/scanner.go#L157-L162)).

I rename this type in https://github.com/github/git-lfs/pull/1595/commits/53f2abb59df469bcf8371580056ef32336274358, and added some docs in https://github.com/github/git-lfs/pull/1595/commits/e16e2de4a11885b5c996f722f24830e0210bb515.